IoTHubBuddy/IoTViewer/AccountManager.cs

```diff
@@ -0,0 +1,362 @@
+        /// <param name="args"></param>
+        private static async void WebAccountInvoked(WebAccountCommand command, WebAccountInvokedArgs args)
+        {
+            if (args.Action == WebAccountAction.Remove)
+            {
+                await SignOutAccountAsync(command.WebAccount);
+            }
+        }
+        /// <summary>
+        /// once the user has logged in, try to get a token for azure management
+        /// </summary>
+        /// <param name="command"></param>
+        private static async void WebAccountProviderCommandInvoked(WebAccountProviderCommand command)
+        {
+            if ((command.WebAccountProvider.Id == MicrosoftProviderId) && (command.WebAccountProvider.Authority == MicrosoftAccountAuthority))
+            {
+                // ClientID is ignored by MSA
+                await AuthenticateWithRequestToken(command.WebAccountProvider, MicrosoftAccountScopeRequested, MicrosoftAccountClientId);
+            }
+            else if ((command.WebAccountProvider.Id == MicrosoftProviderId) && (command.WebAccountProvider.Authority == AzureActiveDirectoryAuthority))
+            {
+                await AuthenticateWithRequestToken(command.WebAccountProvider, AzureActiveDirectoryScopeRequested, AzureActiveDirectoryClientId);
+            }
+
+        }
+        /// <summary>
+        /// attempt to get a token from azure management, if successful then navigate to subscription page
+        /// </summary>
+        /// <param name="provider"></param>
+        /// <param name="scope"></param>
+        /// <param name="client"></param>
+        /// <returns></returns>
+        public static async Task AuthenticateWithRequestToken(WebAccountProvider provider, string scope, string client)
+        {
+            WebTokenRequest request = new WebTokenRequest(provider, scope, client);
+            Debug.WriteLine("provider: " + provider.Id + " scope: " + scope + " client: " + client);
+            if (provider.Id == "https://login.microsoft.com")
+            {
+                request.Properties.Add("resource", "https://management.azure.com/");
+            }
+            WebTokenRequestResult result = await WebAuthenticationCoreManager.RequestTokenAsync(request);
+            if (result.ResponseStatus == WebTokenRequestStatus.Success)
+            {
+                WebAccount account = result.ResponseData[0].WebAccount;
+                StoreWebAccount(account);
+                var sure = result.ResponseData[0];
+                string token = result.ResponseData[0].Token;
+
+                Debug.WriteLine("token: " + token);
+                mp.NavigateToSubscription(token);
+
+            }
+            
+
+        }
+
+        /// <summary>
+        /// store web account in local storage
+        /// NOTE: this will likely change soon
+        /// </summary>
+        /// <param name="account"></param>
+        private static async void StoreWebAccount(WebAccount account)
```

Comment:
why is this function async? I don't see any awaits in it
also, please audit your code for async void functions -- these are bad (event handler have to be like that, but not other methods)